src/components/project/handlers/apply-approved-changeset-to-project.handler.ts

```diff
@@ -35,17 +44,61 @@ export class ApplyApprovedChangesetToProject
       }
       const { id, createdAt, type, financialReportPeriod, ...actualChanges } =
         changes;
-      await this.projectService.update(
-        {
-          id,
-          ...actualChanges,
-          ...(financialReportPeriod ? { financialReportPeriod } : {}),
-        },
-        event.session,
-        undefined,
-        false
+
+      await Promise.all(
+        keys(actualChanges).map(async (key) => {
+          const query = this.db
+            .query()
+            .match(node('node', 'Project', { id }))
+            // Apply previous label to active prop
+            .apply(addPreviousLabel(key, changesetId))
+            // Deactivate active prop
+            .apply(
+              deactivateProperty({
+                key: key as keyof Omit<
+                  DbChanges<TranslationProject | InternshipProject>,
+                  'canDelete'
+                >,
+                resource:
+                  type === ProjectType.Translation
+                    ? TranslationProject
+                    : InternshipProject,
+              })
+            )
+            // Set changed prop to active
+            .apply(activeChangedProp(key, changesetId))
+            .return('node');
+
+          await query.run();
+        })
       );
 
+      // TODO
+      // const query = this.db
+      //   .query()
+      //   .match([
+      //     node('node', 'Project'),
+      //     relation('out', '', 'changeset', { active: true }),
+      //     node('changeset', 'Changeset', { id: changesetId }),
+      //   ])
+      //   .apply(matchProps({ changeset: changesetId, optional: true }))
+      //   .with('props')
+      //   .forEach('prop', 'props', (prop) =>
+      //     prop
+      //       // Deactivate active property
+      //       .apply(
+      //         deactivateProperty({
+      //           key: variable('type(prop)').name as DbChanges<
```

Comment:
@CarsonF How can we get `key(type of prop)` using `variable` here?

Reply:
Using what I said above you could do 
```ts
key: variable('type(relationToProp)')
```
As long as `relationToProp` is what you used in the match above.